drivers/spi/spi_pw.c

```diff
@@ -320,6 +329,153 @@ static void spi_pw_config_clk(const struct device *dev,
 	spi_pw_reg_write(dev, PW_SPI_REG_CTRLR0, ctrlr0);
 }
 
+#ifdef CONFIG_SPI_PW_LPSS_DMA
+static void spi_pw_idma_intr_enable(const struct device *dev, bool enable)
+{
+	uint32_t reg;
+
+	reg = spi_pw_reg_read(dev, PW_SPI_REG_CTRLR1);
+
+	if (enable) {
+		reg |= PW_SPI_IDMA_INTR;
+		spi_pw_reg_write(dev, PW_SPI_REG_CTRLR1, reg);
+	} else {
+		reg &= ~(PW_SPI_IDMA_INTR);
+		spi_pw_reg_write(dev, PW_SPI_REG_CTRLR1, reg);
+	}
+}
+
+static void cb_idma_transfer(const struct device *dma, void *user_data,
+			 uint32_t channel, int status)
+{
+	const struct device *dev = (const struct device *)user_data;
+	struct spi_pw_data *spi = dev->data;
+	int err;
+
+	if (status) {
+		err = 0;
+	} else {
+		err = -EIO;
+	}
```

Comment:
The way you use `err` in this function seems a bit overkill to me. Couldn't you just do the following with your call to spi_context_complete():
```c
spi_context_complete(&spi->ctx, dev, status ? 0 : -EIO);
```

Btw, that still doesn't make the code any more intuitive. You should at least add a code comment as well, explaining why a non-zero status maps to zero error and vice versa.